crates/rspack_core/src/compiler/module_executor/mod.rs

```diff
@@ -65,7 +75,20 @@ impl ModuleExecutor {
     make_artifact.diagnostics = Default::default();
     make_artifact.has_module_graph_change = false;
 
-    make_artifact = update_module_graph(compilation, make_artifact, params)
+    // Modules imported by `importModule` are passively loaded.
+    let mut build_dependencies = self.cutout.cutout_artifact(&mut make_artifact, params);
+    let mut build_dependencies_id = build_dependencies
+      .iter()
+      .map(|(id, _)| *id)
+      .collect::<HashSet<_>>();
+    for mut dep_status in self.request_dep_map.iter_mut() {
+      if build_dependencies_id.contains(&dep_status.id) {
+        dep_status.should_update = true;
+        build_dependencies_id.remove(&dep_status.id);
+      }
+    }
+    build_dependencies.retain(|dep| build_dependencies_id.contains(&dep.0));
```

Comment:
We can merge the build_dependencies processing into a loop

Reply:
I'm not sure how to do it. `build_dependencies` is `HashSet<(DependencyId, Option<Identifier>)>` so we can't remove the elements only by `DependencyId` in the loop.